src/modules/robot/Robot.cpp

```diff
@@ -1698,7 +1677,39 @@ bool Robot::append_milestone(const float target[], float rate_mm_s, unsigned int
     }
 #endif
 
-    DEBUG_PRINTF("distance: %f, aux_move: %d\n", distance, auxilliary_move);
+    if (this->inverse_time_mode) {
+        // in G93/inverse time mode, the feed rate is given as 1/min,
+        // by multiplying it with the distance we get mm/min, the same as in G94
+        feed_rate *= distance;
+    }
+
+    float rate_mm_s = feed_rate / seconds_per_minute;
```

Comment:
as far as I understand, `seconds_per_minute` is used as a scaling factor for the feed overrides. again, I hope that the program is not multi-threaded - the new logic contains several paths where we first multiply and then divide the feed rate by `seconds_per_minute`, which might cause some weird effects if that value is changed meanwhile by another thread.